wire-ios-request-strategy/Sources/Request Strategies/Feature configurations/Payload/MLSFeatureProcessor.swift

```diff
@@ -0,0 +1,59 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireDataModel
+
+// TODO: Add tests
+public protocol MLSFeatureProcessorProtocol {
+
+    func processFeatureConfigChanges(_ mlsFeature: Feature.MLS) async
+
+}
+
+public final class MLSFeatureProcessor: MLSFeatureProcessorProtocol {
+
+    private let mlsClientSyncManager: MLSClientSyncManagerProtocol
+    private let context: NSManagedObjectContext
+
+    public init(
+        coreCryptoProvider: CoreCryptoProviderProtocol,
+        mlsService: any MLSServiceInterface,
+        syncContext: NSManagedObjectContext
+
+    ) {
+        self.mlsClientSyncManager = MLSClientSyncManager(
+            coreCryptoProvider: coreCryptoProvider,
+            mlsService: mlsService)
+        self.context = syncContext
+    }
+
+    public func processFeatureConfigChanges(_ mlsFeature: Feature.MLS) async {
```

Comment:
If this is only called from a single place, I would suggest to just write this method at the call site and avoid wrapping it in another class.

Reply:
removed `MLSFeatureProcessor`